modules/tinymce/src/themes/silver/main/ts/ui/dialog/Tree.ts

```diff
@@ -0,0 +1,177 @@
+import { Behaviour, Button as AlloyButton, Tabstopping, GuiFactory, SimpleSpec, Toggling } from '@ephox/alloy';
+import { Dialog } from '@ephox/bridge';
+import { Fun, Optional } from '@ephox/katamari';
+import { SelectorFind } from '@ephox/sugar';
+
+import { UiFactoryBackstage } from '../../backstage/Backstage';
+import { renderMenuButton } from '../button/MenuButton';
+import * as Icons from '../icons/Icons';
+
+type TreeSpec = Omit<Dialog.Tree, 'type'>;
+type AlloyButtonSpec = Parameters<typeof AlloyButton['sketch']>[0];
+type OnLeafAction = (id: string) => void;
+
+const renderLabel = (text: string ): SimpleSpec => ({
+  dom: {
+    tag: 'span',
+    classes: [ `tox-tree__label` ]
+  },
+  components: [
+    GuiFactory.text(text)
+  ],
+});
+
+const renderItemLabel = (item: Dialog.Leaf, level: number, onLeafAction: OnLeafAction, backstage: UiFactoryBackstage): AlloyButtonSpec => {
+  const internalMenuButton = item.menu.map((btn) => renderMenuButton(btn, 'tox-mbtn', backstage, Optional.none()));
+  const components = [ renderLabel(item.title) ];
+  internalMenuButton.each((btn) => components.push(btn));
+
+  return AlloyButton.sketch({
+    dom: {
+      tag: 'span',
+      classes: [ `tox-tree--item__label`, 'tox-trbtn' ],
+      styles: {
+        'padding-left': `${level * 8}px`,
+        'font-weight': '400',
+      }
+    },
+    components,
+    action: (_button) => {
+      onLeafAction(item.id);
+    },
+    buttonBehaviours: Behaviour.derive([
+      Tabstopping.config({})
+    ])
+  });
+};
+
+const renderIcon = (iconName: string, iconsProvider: Icons.IconProvider, behaviours: Array<Behaviour.NamedConfiguredBehaviour<any, any, any>>): SimpleSpec =>
+  Icons.render(iconName, {
+    tag: 'span',
+    classes: [
+      'tox-tree__icon-wrap',
+      'tox-icon',
+    ],
+    behaviours
+  }, iconsProvider);
+
+const renderIconFromPack = (iconName: string, iconsProvider: Icons.IconProvider): SimpleSpec =>
+  renderIcon(iconName, iconsProvider, []);
+
+const renderDirectoryLabel = (directory: Dialog.Directory, level: number, backstage: UiFactoryBackstage): AlloyButtonSpec => {
+  const internalMenuButton = directory.menu.map((btn) => renderMenuButton(btn, 'tox-mbtn', backstage, Optional.none()));
+  const components: SimpleSpec[] = [
+    {
+      dom: {
+        tag: 'div',
+        classes: [ 'tox-chevron' ]
+      },
+      components: [
+        renderIconFromPack('chevron-right', backstage.shared.providers.icons),
+      ]
+    },
+    {
+      dom: {
+        tag: 'span',
+        styles: {
+          'padding-left': '4px'
+        }
+      },
+      components: [ renderLabel(directory.title) ]
+    }
+  ];
+  internalMenuButton.each((btn) => {
+    components.push(btn);
+  });
+  return AlloyButton.sketch({
+
+    dom: {
+      tag: 'div',
+      classes: [ `tox-tree--directory__label`, 'tox-trbtn' ],
+      styles: {
+        'padding-left': `${level * 8}px`,
+        'font-weight': '700',
```

Comment:
This should also be in the skin css.

Reply:
The padding might be hard to not have as a style. Nesting them to just make them skinnable might be messy. But as much as possible should be skinnable with css.

Reply:
It is possible. I just pushed a fix.